src/test/kotlin/com/shionfujie/rx/util/RecordingObserver.kt

```diff
@@ -0,0 +1,84 @@
+package com.shionfujie.rx.util
+
+import io.reactivex.observers.DisposableObserver
+import org.junit.jupiter.api.Assertions.assertTrue
+import java.util.concurrent.LinkedBlockingDeque
+import java.util.concurrent.TimeUnit
+
+class RecordingObserver<T> : DisposableObserver<T>() {
```

Comment:
I've ~~stolen~~ learned this from https://github.com/JakeWharton/RxBinding/blob/master/testing-utils/src/main/java/com/jakewharton/rxbinding3/RecordingObserver.java